python/lsst/ts/standardscripts/maintel/m1m3/check_hardpoint.py

```diff
@@ -0,0 +1,89 @@
+# This file is part of ts_standardscripts
+#
+# Developed for the LSST Telescope and Site Systems.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ["CheckHardpoint"]
+
+from lsst.ts.idl.enums.MTM1M3 import HardpointTest
+from lsst.ts.observatory.control.maintel.mtcs import MTCS, MTCSUsages
+
+from lsst.ts import salobj
+
+
+class CheckHardpoint(salobj.BaseScript):
+    """Check M1M3 Individual hardpoint breakaway.
+
+    Parameters
+    ----------
+    index : `int`
+        Index of Script SAL component.
+
+    Notes
+    -----
+    **Checkpoints**
+
+    - "Testing hardpoint {id}.": Check hardpoint.
+    - "Hardpoint breakaway check complete.": Check complete.
+
+    """
+
+    def __init__(self, index, add_remotes: bool = True):
+        super().__init__(index=index, descr="Check M1M3 Hardpoint")
+
+        self.mtcs = MTCS(
+            domain=self.domain,
+            intended_usage=None if add_remotes else MTCSUsages.DryTest,
+            log=self.log,
+        )
+
+    @classmethod
+    def get_schema(cls):
+        return None
+
+    async def configure(self, config):
+        # This script does not require any configuration.
+        pass
+
+    def set_metadata(self, metadata):
+        metadata.duration = 180.0
+
+    async def wait_hardpoint_test_ok(self):
+        """Wait for the laser to warm up and be ready to be used."""
+
+        self.log.info("Checking if the bump test has passed.")
+
+        self.mtcs.rem.mtm1m3.evt_hardpointTestStatus.flush()
+        test_status = await self.mtcs.rem.mtm1m3.evt_hardpointTestStatus.aget(
+            timeout=self.timeout_std,
+        )
+
+        while test_status.status != HardpointTest.PASSED:
+            test_status = await self.mtcs.rem.mtm1m3.evt_hardpointTestStatus.next(
+                flush=False,
+                timeout=self.timeout_check,
+            )
+
+    async def run(self):
+        for id in range(7):
```

Comment:
I think you should probably do some basic checks that the M1M3 is in ENABLED state and I also think you need to put it in engineering mode with the:

```
await self.mtcs.rem.mtm1m3.cmd_enterEngineering.start(timeout=5)
```

command.

And I think you need to take it out of engineering mode afterwards. However, this last part is probably best done in the `cleanup` method, which is something that the script always run at the end, e.g. see [this](https://github.com/lsst-ts/ts_standardscripts/blob/559f76acb992564353687a0cdbbebd361bd58985/python/lsst/ts/standardscripts/base_track_target.py#L371).

Reply:
Keep in mind cmd_enterEngineering will end with an error if the system is already in an engineering state.

Reply:
@tribeiro should i then add an assert to check that we are not already in eingineering state?